atom/browser/api/event_subscriber.h

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2017 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_BROWSER_API_EVENT_SUBSCRIBER_H_
+#define ATOM_BROWSER_API_EVENT_SUBSCRIBER_H_
+
+#include <map>
+#include <string>
+
+#include "atom/common/api/event_emitter_caller.h"
+#include "base/synchronization/lock.h"
+#include "content/public/browser/browser_thread.h"
+#include "native_mate/native_mate/arguments.h"
+
+namespace mate {
+
+namespace internal {
+
+class EventSubscriberBase {
```

Comment:
> I think EventSubscriberBase can be moved to native-mate

:+1: 

> and can reuse some utilities from there, for example the CallbackHolderBase for JSHandlerData. What do you think ?

Sure, but the main reason I implemented this class is because I couldn't make it work with the existing infrastructure. In particular, I tried to use the existing converters to pass a `base::Callback` to the javascript `EventEmitter.on` method but it didn't work as expected. IIRC it only executed the callback once.

I can try to sort through the issues and refactor to use more of the existing helpers but it will take some time. Can I do this (moving to native-mate + refactoring) after this PR is merged?

Reply:
> IIRC it only executed the callback once.

Yup, we should probably try to use the new `base::Callback` variant, [base::RepeatingCallback](https://chromium.googlesource.com/chromium/src/+/master/docs/callback.md). Can be done separately, not related to the PR.  

> I can try to sort through the issues and refactor to use more of the existing helpers but it will take some time. Can I do this (moving to native-mate + refactoring) after this PR is merged?

Sure, sounds good to me.